### PR TITLE
hyundai: WHL_SPD11 checksum

### PR DESCRIPTION
**Pull request by gregjhogan**
note that they excluded the counter from the checksum

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can you also try adding to the hyundai legacy checks and run the openpilot tests?

**Comment by adeebshihadeh** [comment]
Looks like we'll have to leave this out of the legacy safety mode.

**Comment by gregjhogan** [comment]
should be good to merge now

File: board/safety/safety_hyundai.h [hunk 1]
```diff
@@ -77,15 +77,31 @@ static uint8_t hyundai_compute_checksum(CAN_FIFOMailBox_TypeDef *to_push) {
...
+        if (((i != 1) || (j < 6)) && ((i != 3) || (j < 6)) && ((i != 5) || (j < 6)) && ((i != 7) || (j < 6))) {
+          bit = (b >> (uint8_t)j) & 1U;
+        }
+        chksum += bit;
+      }
     }
-    chksum += (b % 16U) + (b / 16U);
+    chksum = (chksum ^ 9U) & 15U;
+  } else {
+    // same algorithm, but checksum is in a different place
```
Comment (adeebshihadeh) [inline board/safety/safety_hyundai.h:93]: this comment is ambiguous now, can you update it?

Comment (gregjhogan) [reply board/safety/safety_hyundai.h]: done

**Pull request merged**